components/Header/Header.tsx

```diff
@@ -9,40 +9,62 @@ import HeaderCTA from "./HeaderCTA";
 import styles from "./Header.module.css";
 import Magnifier from "./assets/magnify.svg?react";
 import Link from "components/Link";
+import { EventBanner, EventProps } from "components/EventBanner";
+import eventData from "../../public/events.json";
 
 const Header = () => {
+  const [events] = useState<EventProps[]>(eventData);
+  const [screenWidth, setScreenWidth] = useState(900);
   const [isNavigationVisible, setIsNavigationVisible] =
     useState<boolean>(false);
   const toggleNavigaton = useCallback(() => {
     setIsNavigationVisible((value) => !value);
     blockBodyScroll(isNavigationVisible);
   }, [isNavigationVisible]);
 
+  const handleResize = () => {
+    setScreenWidth(window.innerWidth);
+  };
+
+  useEffect(() => {
+    window.addEventListener("resize", handleResize);
+
+    return () => {
+      window.removeEventListener("resize", handleResize);
+    };
+  }, [screenWidth]);
   return (
-    <header className={styles.wrapper}>
-      <a href="/" className={styles["logo-link"]}>
-        <Logo />
-      </a>
-      <div className={styles.searchLink}>
-        <NavSearch testid="mobile-search" aria-details="Search website" />
-      </div>
-      <HeadlessButton
-        onClick={toggleNavigaton}
-        className={styles.hamburger}
-        data-testid="hamburger"
-        aria-details="Toggle Main navigation"
-      >
-        <Icon name={isNavigationVisible ? "close" : "hamburger"} size="md" />
-      </HeadlessButton>
-      <div
-        className={cn(styles.content, {
-          [styles.visible]: isNavigationVisible,
-        })}
+    <>
+      <EventBanner events={events} />
+      <header
+        style={{ top: !!events.length && screenWidth < 900 ? "48px" : 0 }}
+        className={styles.wrapper}
       >
-        <Menu />
-        <HeaderCTA />
-      </div>
-    </header>
+        <a href="/" className={styles["logo-link"]}>
+          <Logo />
+        </a>
+        <div className={styles.searchLink}>
+          <NavSearch testid="mobile-search" aria-details="Search website" />
+        </div>
+        <HeadlessButton
+          onClick={toggleNavigaton}
+          className={styles.hamburger}
+          data-testid="hamburger"
+          aria-details="Toggle Main navigation"
+        >
+          <Icon name={isNavigationVisible ? "close" : "hamburger"} size="md" />
+        </HeadlessButton>
+        <div
+          className={cn(styles.content, {
+            [styles.visible]: isNavigationVisible,
+          })}
+          style={{ top: !!events ? "96px" : "48px" }}
```

Comment:
What is the difference between `!!events` and `!!events.length` used earlier in the code?

Reply:
Line 61 should have !!events.length also. 

Difference between the two is that if the built Sanity query doesn't have events to return it returns an empty array. This still counts as there were events and rendering would be incorrect. Events.length checks if there is even one event inside the array and if it's empty it counts as no events.